File: trivially/src/main/java/trivially/ep/m7i2/Add.java
package trivially.ep.m7i2;

public interface Add<V> extends trivially.ep.m7.Add<V>, trivially.ep.i2.Add<V>, Exp<V> {

    Exp<V> getLeft();
    Exp<V> getRight();

    default Exp<V> simplify() {
        return this.convert(trivially.ep.m7.Add.super.simplify());
    }

    default Exp<V> multby(trivially.ep.Exp<V> other) {
        return this.mult(this, convert(other));
    }

    default Exp<V> powby(trivially.ep.Exp<V> other) {
        return this.power(this, convert(other));
    }
}

File: coco_csharp/CoCo/EP/M7I2/Add.cs
namespace EP.M7I2 {
    public interface Add<FT> : Exp<FT>, EP.M7.Add<FT>, EP.I1.Add<FT> {
        EP.Exp<FT> EP.I1.Exp<FT>.multby(EP.Exp<FT> other) {
            return mult(this, convert(other));
        }
        EP.Exp<FT> EP.M7.Exp<FT>.powby(EP.Exp<FT> other) {
            return power(this, convert(other));
        }
    }
}